domain/src/main/java/life/qbic/domain/usermanagement/registration/RegisterUserOutput.java

```diff
@@ -20,5 +22,7 @@ public interface RegisterUserOutput {
    * @param reason the reason for the user registration failure
    * @since 1.0.0
    */
+  void onFailure(UserRegistrationException exception);
```

Comment:
```suggestion
  void onUserRegistrationFailed(UserRegistrationException exception);
```

Reply:
`onFailure` and `onSuccess` are not really speaking names of interface methods. In the case that a class implements multiple interfaces you would have to backtrace which interface provided the methods to read the context `RegisterUserOutput::onFailure`. In case multiple interfaces use the same method name no distinction is possible anymore.